components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java

```diff
@@ -159,9 +159,9 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
     public Map<String, Object> populateStandardClaims(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementException {
 
         //generating expiring timestamp
-        long currentTime = System.currentTimeMillis();
+        long currentTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
         // jwtTokenInfoDTO.getExpirationTime() gives the token validity time given when the token is generated.
-        long expireIn = TimeUnit.MILLISECONDS.toSeconds(currentTime) + jwtTokenInfoDTO.getExpirationTime();
+        long expireIn = currentTime + jwtTokenInfoDTO.getExpirationTime();
```

Comment:
something is missing here. is `jwtTokenInfoDTO.getExpirationTime()` in seconds or miliseconds?

Reply:
According to the following implementation 
jwtTokenInfoDTO.setExpirationTime(getSecondsTillExpiry(tokReqMsgCtx.getValidityPeriod()));
it's in second.

Reply:
hmm does that mean we added seconds to milliseconds previously? :-/

Reply:
No. Ealier currentTime is in millisecond and convert to seconds before adding, Refer following. 
![image](https://user-images.githubusercontent.com/42195796/49269289-3c5a1180-f48a-11e8-8cfe-69235a4ff2c3.png)

Reply:
ok, I missed the `toSeconds` part. Now there's another question. What's the difference between now and then?

Reply:
'iat' is set from the currentTime value.Ealier it was in millisecond(currentTime). From this PR it changed to seconds.